ProcessMaker/processmaker
# FOUR-6601: Refactor settings

## Environment Setup
- Set core to this PR's branch: [FOUR-6601-develop](https://github.com/ProcessMaker/processmaker/tree/FOUR-6601-develop)
- Set connector-send-email to [FOUR-6601-develop](https://github.com/ProcessMaker/connector-send-email/pull/216)
- Set package-collections [FOUR-6601-develop](https://github.com/ProcessMaker/package-collections/pull/323)
- Set package-auth to [FOUR-6601-develop](https://github.com/ProcessMaker/package-auth/pull/65)
- Set package-ellucian-ethos to [FOUR-6601-develop](https://github.com/ProcessMaker/package-ellucian-ethos/pull/71)

## Issue & Reproduction Steps
We used to utilize a tagged Redis cache to store settings to reduce the number of queries to the database every time the app is loaded. Over time, we've noticed the cache actually creates too many network calls to Redis (it makes a network call everytime a cached setting is called) and it uses a lot more memory by each PHP process. We also had a `settings()` function helper which complicates things as all settings were also in the global app config (accessed using the `config()` function helper).

This PR refactors the settings so they can be easily accessed through the global app configuration. This will reduce the duplicative nature of the `settings()` helper and simplify the app config caching process. 

## Solution
If the app configuration is cached when a setting is saved or updated, the config will automatically be cleared and re-cached. Horizon will also be restarted when the configuration is re-cached to ensure it picks up the updated config values.

## Refactoring settings usage in packages
Multiple packages implement and use settings, however this refactor deprecates the `settings()`, `flush_settings()`, and the `cache_settings()` function helpers. There are also other changes which need to be made to specific place in the below packages to ensure the settings are correctly implemented.

## How to Test
Extensive testing will be required to ensure all PRs cover all use cases of settings, ensuring they are updated (even when the configuration is cached).

## Related Tickets & Packages
- [PR for connector-send-email](https://github.com/ProcessMaker/connector-send-email/pull/216)
- [PR for package-collections](https://github.com/ProcessMaker/package-collections/pull/323)
- [PR for package-auth](https://github.com/ProcessMaker/package-auth/pull/65)
- [PR for package-ellucian-ethos](https://github.com/ProcessMaker/package-ellucian-ethos/pull/71)
- [Ticket FOUR-6601](https://processmaker.atlassian.net/browse/FOUR-6601)

## Code Review Checklist
